### PR TITLE
ISSUE-213 Provide Guice for JwtPrivateKeyProvider

### PR DESCRIPTION
**Pull request by vttranlina**
Ref https://github.com/linagora/tmail-backend/issues/213

### CODE REVIEW

**Comment by vttranlina** [comment]
As conservations at https://github.com/linagora/tmail-backend/pull/234#issuecomment-923872071
I tried to create
- LinagoraJMAPDraftConfiguration (adapt from JMAPDraftConfiguration)
- LinagoraJMAPModule (adapt from JMAPModule)
- LinagoraJMAPServerModule (adapt from JMAPServerModule)
...

But I realized that have many classes depend on `JMAPDraftConfiguration` (James and tmail), so I think I should not override it.
With the way in PR, IMO all will simpler

File: tmail-backend/guice/encrypted-mailbox/src/test/java/com/linagora/tmail/module/LinagoraTestJMAPServerModule.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import org.apache.james.modules.TestJMAPServerModule;
+
+import com.google.inject.Provides;
+import com.linagora.tmail.james.jmap.jwt.JwtPrivateKeyConfiguration;
+import com.linagora.tmail.james.jmap.jwt.JwtPrivateKeyProvider;
+
+import scala.jdk.javaapi.OptionConverters;
+
+public class LinagoraTestJMAPServerModule extends TestJMAPServerModule {
```
Comment (Arsnael) [inline tmail-backend/guice/encrypted-mailbox/src/test/java/com/linagora/tmail/module/LinagoraTestJMAPServerModule.java:15]: Why putting this in the encrypted-mailbox guice module? It has nothing to do with the encrypted mailbox right?

Comment (vttranlina) [reply tmail-backend/guice/encrypted-mailbox/src/test/java/com/linagora/tmail/module/LinagoraTestJMAPServerModule.java:15]: I thought `encrypted-mailbox` is regarding about all tmail-mailbox.
I will create a new submodule `guice/jmap` for it

**Review by chibenwa — COMMENTED** [review]
Nothing to add

**Review by chibenwa — APPROVED** [review]

**Comment by vttranlina** [comment]
@Arsnael May you show me why ci fail? 
My access Denied

**Comment by chibenwa** [comment]
Deployment tests?

**Comment by Arsnael** [comment]
> @Arsnael May you show me why ci fail?
> My access Denied

Denied again? Thought that was fixed last time?

**Comment by Arsnael** [comment]
Looks like a BS error, restarted it

**Review by Arsnael — APPROVED** [review]

**Comment by vttranlina** [comment]
> > @Arsnael May you show me why ci fail?
> > My access Denied
> 
> Denied again? Thought that was fixed last time?

Yes, I have permission on Github, but not on Jenkin 

(screenshot from Incognito window)
image

**Comment by vttranlina** [comment]
Yeah, all green

**Comment by Arsnael** [comment]
Seems the plugin is a bit sh*t regarding permissions so I might need to find an other way... I do have an idea but I need a slot of time for checking it.

**Pull request merged**